packages/remix-dev/compiler/plugins/serverRouteModulesPlugin.ts

```diff
@@ -0,0 +1,48 @@
+import * as path from "path";
+import * as fsp from "fs/promises";
+
+import type esbuild from "esbuild";
+
+import type { RemixConfig } from "../../config";
+import { getLoaderForFile } from "../loaders";
+
+/**
+ * This plugin loads route modules for the server build.
```

Comment:
"...and helps prevent errors when adding new files in development so esbuild doesn't complain about empty modules"